pkg/volume/util/nestedpendingoperations/nestedpendingoperations.go

```diff
@@ -128,15 +127,20 @@ func (grm *nestedPendingOperations) Run(
 			})
 	}
 
-	go func() (err error) {
+	go func() (eventErr, detailedErr error) {
 		// Handle unhandled panics (very unlikely)
 		defer k8sRuntime.HandleCrash()
 		// Handle completion of and error, if any, from operationFunc()
-		defer grm.operationComplete(volumeName, podName, &err)
-		defer operationCompleteFunc(err)
+		defer grm.operationComplete(volumeName, podName, &detailedErr)
+		if generatedOperations.CompleteFunc != nil {
+			defer generatedOperations.CompleteFunc(&detailedErr)
+		}
+		if generatedOperations.EventRecorderFunc != nil {
+			defer generatedOperations.EventRecorderFunc(&eventErr)
+		}
 		// Handle panic, if any, from operationFunc()
-		defer k8sRuntime.RecoverFromPanic(&err)
-		return operationFunc()
+		defer k8sRuntime.RecoverFromPanic(&detailedErr)
```

Comment:
same as before - I do not see why we are passing address of an interface in those places. @davidz627 can you explain?

Reply:
This was added to address a bug I found while writing this PR. Previously the CompleteFuncs were not logging error at all.

In the previous version of the function take a look at the line:
defer operationCompleteFunc(err)
Because Go evaluates "defer"s variables at runtime, we see that the value of "err" will always be nil in this version.

The fixed version gives this operationCompleteFunction a pointer to err. Therefore when the OperationFunc completes execution, the RecoverFromPanic function will actually write a value to the location of the err pointer in case of error. Then the CompleteFunc will be able to use this error for logging or its other functions by inspecting the value written at the pointer location.

Reply:
Hope that helps! Let me know if there is any flaw in my reasoning.

I also did some testing to confirm the errors were not being logged previously, and are now being logged successfully after this change.